rust/lance/src/dataset/scanner.rs

```diff
@@ -643,7 +652,7 @@ impl Scanner {
     async fn create_plan(&self) -> Result<Arc<dyn ExecutionPlan>> {
         // Scalar indices are only used when prefiltering
         // TODO: Should we use them when postfiltering if there is no vector search?
-        let use_scalar_index = self.prefilter;
+        let use_scalar_index = self.prefilter || self.nearest.is_none();
```

Comment:
This condition was preventing scalar indices from being used in scans.

Reply:
Why do we want to do that?

Reply:
You think we shouldn't?

I was thinking a user might expect a query like:

```python
dataset.to_table(filter="id = 42")
```

To use a scalar index on `id` if it was available. But I suppose in other cases it might not be desirable, like if the filter isn't very selective. Is that what you are getting at?

Reply:
No, sorry, I was reading your statement backwards.  I thought you were disabling scalar indices.  I do think I need to add a "use_scalar_index" field soon though because you're right, there will be queries where the scalar index hurts rather than helps.

For now though, this change is good.